### DISCUSSION TITLE
更新日志 - 22.03.4 - 20230714

### DISCUSSION THREAD

**Discussion by jjm2473** [istoreos/istoreos#878] [Announcements]
# 应用：
* QuickStart：“终端”按钮支持TTYD的用户自定义端口和SSL配置
* iStore：新上架nps，airplay2
* 动态DNS：集成dnspod，3322.net

# 设备：
* x86_64：加回风扇控制界面（由于BIOS差异，不保证可用）
* R5S/R5C：修复LED灯
* H69K：修复初次启动可能没打开WIFI的问题

**Comment by xixiluhui520** [comment]
老大什么时候放包呢，期待

**Reply by xixiluhui520** [reply]
已下载

**Comment by xixiluhui520** [comment]
还是跟0707版一样，LAN口没有输出了

**Comment by xixiluhui520** [comment]
设备重启后，好像连接不上了，不知道系统坏没有

**Reply by jjm2473** [reply]
啥设备啊，重做系统还有问题吗

**Reply by xixiluhui520** [reply]
笔记本，嗯更新后LAN口无连接，现在重启后设备后远程连接不上了，要晚上回家后才知道怎么回事了.

**Reply by xixiluhui520** [reply]
现在连接终于上了.看来设备系统启动用了10几分钟，重启后路由器依然无连接

**Reply by xixiluhui520** [reply]
老大，我应该怎么办，只能用0609版了？

**Reply by ymmzjx** [reply]
x86的风扇控制加在哪了?一直找不到

**Reply by jjm2473** [reply]
@ymmzjx 不是所有机型都有的，要看bios有没有支持内核的风扇功能

**Comment by TIMhao** [comment]
我也是x86设备，pve安装的，更新正常，但是iStore OS里的终端打不开了，更新之前可以，默认端口7681我通过本地其他机器ping了一下不通，开启的passwall的HTTP 监听端口7890端口是通的，应该是不是防火墙问题，可能还是终端程序的问题

**Reply by jjm2473** [reply]
确实有这个问题，我重新编译下ttyd

**Reply by wcyws** [reply]
发现相同问题....真的有点不敢更新了，每次更新基本上都会有问题，上一次是docker和smb，最后是直接把rom里面的数据覆盖掉现有数据才恢复正常...能稳定用还是稳定用吧

**Reply by jjm2473** [reply]
@wcyws 你这种情况肯定是因为以前自己更新过samba，docker的话，则是之前的配置文件不兼容而已

**Reply by xixiluhui520** [reply]
我是x86，固态硬盘安装，0707和0714都是LAN口无输出，所有接口都是无法编辑会报错，这次0714更为严重一点，重启后要很久才能远程连接，显示局域网连接依然是：连接0

**Reply by jjm2473** [reply]
@wcyws 而且你要是把rom里面的数据都复制出来，那以后就不用更新了。rom里面的东西就是固件原始的文件，要是复制出来，那以后更新也是被你之前复制的文件覆盖，也就没更新了。不要整这些骚操作，真要复制rom的文件，也只能复制 /rom/etc/config 里面的

**Reply by wcyws** [reply]
是的，zerotier和lan都需要使用samba，但接口只能选一个，只修改了强制绑定接口选项，windows可以看到samba，但点击进去就是拒绝连接...怎么调试配置都无法使用...最后把rom下所有samba文件覆盖过去才恢复正常...原因也找不到
docker是直接没有下拉菜单了，只有一个docker的图标
这次是发现没有terminal了，7681端口拒绝访问...欸...

**Reply by wcyws** [reply]
啊？ 我以为是rom也会随着刷写固件来update呢......

**Reply by jjm2473** [reply]
@wcyws ttyd的问题是原版就有的，又不是我们引入的bug，主要原因是依赖的ws库有版本校验，如果ws库编译了，ttyd却没重新编译，就会启动失败，我也觉得莫名其妙，毕竟什么都没改

**Reply by wcyws** [reply]
抱歉抱歉，你们更新是辛苦的，我完全没有责怪的意思哈，就是每次更新自己的设备老出问题吐槽一下了，其实本人也没改什么东西了，安装了4个插件，5个docker容器而已，但依然还是不够稳定。根据你之前的回复，我后面可能不选择更新了，请问有没有稳定的版本推荐呀？ 设备是畅网的 x86_64

**Reply by wcyws** [reply]
@jjm2473

**Reply by jjm2473** [reply]
@wcyws rom是会随着固件更新啊，但是你复制出来以后，对应的文件被覆盖掉了，也就是overlay里面是你复制的，那以后rom更新了也没用，overaly才是上层。算了，解释再多也没用，自己去了解overalyfs的机制，去了解openwrt怎么实现rom跟用户数据分离，怎么实现恢复出厂设置

**Reply by wcyws** [reply]
@jjm2473  抱歉再打扰下，刚去查了一下，/rom是下层，然后同路径的是上层，下层包含系统基础配置，上层则是用户配置，overlayfs是把上下层虚拟在一起，先再上层查找，找不到再去下层。假设我samba出问题了，/bin/samba这个文件损坏，那么我从/rom/bin/samba把数据copy出来覆盖掉/bin/samba这样samba就恢复正常啦，我简单理解就是假设用户在上层配错了配置导致无法启动某些应用，从/rom同路径取出文件，再覆盖就可以恢复到初始设置了，这样理解对吗？然后就是每次更新/rom下面的文件其实是不会更新的，在你做U盘的那版系统已经确定了，所以你的意思是从/rom拷贝出来的数据其实是老版本的，所以无效的吗？

**Reply by jjm2473** [reply]
不是，rom会更新，但是你复制出来到overlay的文件优先级更高。
例如rom里面的a文件版本1，被你复制到overlay了，那overlay里也有a文件，且版本是1。固件升级之后rom的a文件升级到了版本2，但是系统还是先找到overlay的a文件，这个文件还是版本1，那rom的更新不就没用上了吗。

**Reply by wcyws** [reply]
@jjm2473 懂了！！ 非常感谢，因为每次遇到程序问题搞不定，然后又不会debug的时候，我都是直接从rom 拷出来的.... 解决问题很快很方便..... 但是拷出来的自带版本属性导致无法用上最新的，那确实不好，这样的话，我应该怎么处理呢，我又不想重装整个...有些插件我不懂，找不全配置，重装手动调配要很久...  从理解上看，只要把overlay里面的a文件干掉，系统应该会自动跑去下层rom里面找？比如samba坏了，那我直接干掉/overlay/bin/samba其实就好了对叭？ 或者说，我直接删掉/bin/samba，然后做个link到rom里面也可以？/bin/samba -> /rom/bin/samba ?  感谢你的解答，我可能要把copy的文件找出来，在处理一遍了_(:з」∠)_

**Reply by jjm2473** [reply]
如果是因为自己升级系统软件包导致的故障，可以使用istore里的systools修复系统软件恢复。如果不是，那就只能恢复出厂设置了。

> 从理解上看，只要把overlay里面的a文件干掉，系统应该会自动跑去下层rom里面找？比如samba坏了，那我直接干掉/overlay/bin/samba其实就好了对叭

这个理解没错，不过文件对应的位置是/overlay/upper/

如果你只是想保留配置，我建议你还不如备份配置后恢复出厂设置，然后再导入配置，最多就再安装点插件而已，安装插件再怎么样也比你找文件简单。而且istore还有备份插件列表的功能，如果是istore安装的插件可以一键恢复。

**Reply by wcyws** [reply]
我直接找到/overlay/upper/下面的所有文件，然后对比，如果rom下面存在，那我就删  命令是  cd /overlay/upper/; find . -regex '.*' | awk '{a=$0; b="/rom/"$0; if ((getline<b)>0) {system("rm -rf "a)}}'. 希望能帮助到跟我遇到一样问题的小伙伴，再次感谢大佬的回答

**Comment by a447948814** [comment]
更新后首页的终端按钮咋没了

**Reply by jjm2473** [reply]
ttyd启动失败了，等更新

**Comment by TIMhao** [comment]
> 确实有这个问题，我重新编译下ttyd

好滴

**Comment by xixiluhui520** [comment]
看样子好像只有我是主路由模式.........，老大能不能解决一下

**Reply by jjm2473** [reply]
你别刷屏了好吧，去开issue，把日志贴一下。

**Reply by xixiluhui520** [reply]
好.能进系统了马上贴

**Reply by jjm2473** [reply]
大家都是主路由模式，旁路由反而是少数

**Comment by jjm2473** [comment]
更新：
固件已经重新上传，如果ttyd启动失败的话就更新下固件，rk35xx机型应该不受影响（ttyd之前就已经重新编译过）。

---
部分机型的ttyd故障，无法启动：
```
Fri Jul 14 15:23:55 2023 daemon.err ttyd[20431]: lws_create_context: failed to load evlib_uv
Fri Jul 14 15:23:55 2023 daemon.err ttyd[20431]: libwebsockets context creation failed
```
ttyd和依赖库都没更新，尚不明确这种bug怎么产生的。
根据之前经验，需要重新编译ttyd。

**Reply by a447948814** [reply]
加油！我是软路由，win的hypv安装的istore旁路由用，等更新。

**Reply by a447948814** [reply]
可以了  已更新iStoreOS 22.03.4 2023071416，终端正常了

**Comment by SN-STAR** [comment]
ipv6测试从最开始那版就一直不通过，怎么设置呢？

**Reply by jjm2473** [reply]
自己搜下issue和discussions

**Comment by luoyeguizhong** [comment]
更新后下载器qBittorrent，Transmission，Aria2都不能用了，tracker服务器都连不上了，重新安装也不行

**Reply by scrymile** [reply]
一样!!!!!!!根本用不了了，哭了，不知道该等谁更新

**Reply by luoyeguizhong** [reply]
感觉是路由器的问题，不是qb的问题，我上面的openclash开代理，在外面用电脑连接，百度查IP是家里软路由的IP，电脑上的qb就用不了了，tracker服务器连不上

**Reply by whispermix** [reply]
我更新后都能正常使用了啊，重新安装就好了

**Comment by wolfbinhe** [comment]
友善R5S更新完，LAN口指示灯修复正常了，但是WAN口的灯还是不亮。。。 大佬看看是不是还没有修复

**Reply by whispermix** [reply]
我的正常亮没问题重启就好了

**Comment by picc745514** [comment]
上架的两个，nps都是客户端啊

**Comment by xixiluhui520** [comment]
老大我想咨询一下，如果我恢复出厂设置后， docker的容器.镜像数据还在不在

**Reply by jjm2473** [reply]
如果你是放在硬盘或者非系统分区，那还是可以恢复的，只要之后把硬盘挂载到相同的位置，并且把docker的数据目录改成跟之前一样。
如果你是放在系统分区里，默认情况下就是，那就先在首页把docker迁移到硬盘上。

**Reply by xixiluhui520** [reply]
/overlay/upper/opt/docker
我固态硬盘安装完系统后直接把多余空间全部划给了分区2

**Reply by xixiluhui520** [reply]
看来我要插一个U盘做迁移了，谢谢！

**Reply by xixiluhui520** [reply]
老大有没有什么方法在我硬盘现有的基础上不插U盘，把docker独立出来呀

**Reply by jjm2473** [reply]
你的硬盘够大，分个区不就行了，然后把docker迁到这个分区。

**Reply by xixiluhui520** [reply]
docker迁移到别的分区，我还不会........老大能不能教教我

**Reply by xixiluhui520** [reply]
难道我分好区之后一样利用首页的docker迁移向导来 完成？如果是这样老大做的系统就太方便啦！！

**Reply by xixiluhui520** [reply]
还有一个疑惑老大，这个docker迁移向导的工作方式是剪切的还是复制的？

**Comment by qsect** [comment]
什么时候修复r66s的默认wan口拨号错误问题，必须将wan和lan更换之后，才能正常使用，否则直接使用默认的wan无法进行拨号

**Reply by jjm2473** [reply]
好像是不能自适应百兆，你还是先检查下是不是网线不行，为什么没千兆

**Comment by xixiluhui520** [comment]
给老大汇报一下，我昨天晚上用0714新版本重新的刷了一下系统（docker迁移已完成）。结果是：LAN口无法连接路由器，所以不能进入WEB管理界面。情急之下又只能刷入0609版本，之后正常经过LAN口进入管理界面。
再然后首页点击更新系统，再恢复了出厂设置，再恢复备份，一套下来终于成功无问题的升级到了最新版的0714版了，也就是这个版本我的设备必须要恢复出厂设置。感觉是这个版本不恢复出厂设置DHCP不工作，不分配IP地址似的。
我是X64平台！

**Reply by xixiluhui520** [reply]
刷入系统方法：PE微系统用phydiskwrite.exe直接刷入固态硬盘

**Reply by jjm2473** [reply]
怎么可能呢，你还是检查下是不是有多个系统吧，例如u盘和硬盘都有系统，自己都不知道启动的是哪一个

**Reply by xixiluhui520** [reply]
没有，我的U盘没有系统，我的固态硬盘上也只有你的系统，我安装之前都是把整个硬盘直接删除整个分区，再刷系统，再PE里面把多余的空间分配好，这次我听你的专门给/docker分了一个独立区，进系统还原备份后再重新挂载/docker分区

**Reply by xixiluhui520** [reply]
<html>
<body>


主机名 | iStoreOS
-- | --
型号 | LENOVO 20131
架构 | Intel(R) Core(TM) i7-3632QM CPU @ 2.20GHz
目标平台 | x86/64
固件版本 | iStoreOS 22.03.4 2023071416 / LuCI istoreos-22.03 branch git-23.103.23275-0193674
内核版本 | 5.10.176
本地时间 | 2023-07-19 13:43:09
运行时间 | 14h 42m 44s
平均负载 | 0.02, 0.08, 0.08


</body>
</html>

**Reply by xixiluhui520** [reply]
现在是正常了，但是新版固件对于我的固件来说的确是有BUG

**Reply by xixiluhui520** [reply]
我的硬件

**Reply by xixiluhui520** [reply]
[istoreos-22.03.4-2023071416-x86-64-squashfs-combined.img.gz](https://fw0.koolcenter.com/iStoreOS/x86_64/istoreos-22.03.4-2023071416-x86-64-squashfs-combined.img.gz)这个固件我直接phydiskwrite.exe刷入后LAN口是连接不了的，刷了两次。都不行。当然我说的是对于我的设备硬件

**Reply by xixiluhui520** [reply]
只能刷0609后再升级，再然后恢复出厂设置再恢复备份才能正常食用了！

**Reply by jjm2473** [reply]
你没用efi固件？

**Reply by xixiluhui520** [reply]
efi固件不支持pe里面直接分区。。。。

**Reply by xixiluhui520** [reply]
通用固件不行吗

**Reply by jjm2473** [reply]
谁知道你的bios和硬件什么情况，能用efi就用efi。分区功能在istoreos的后台就有，还要啥pe。就算是pe，现在都什么年代了，还不支持gpt？

**Reply by xixiluhui520** [reply]
好的老大我以后用efi版本，我的主板和BIOS都支持efi的。

**Comment by pingjiao99** [comment]
管控里的功能都无效，被限制的设备还能正常上网

**Comment by gzxiexl** [comment]
申请iStore支持tailscale！！！

**Reply by jjm2473** [reply]
什么时候不支持了？tailscale依赖的是wireguard，一直都有集成。

**Reply by ZRP1107** [reply]
wget https://github.com/adyanth/openwrt-tailscale-enabler/releases/download/v1.36.1-fb2f6cf-autoupdate/openwrt-tailscale-enabler-v1.36.1-fb2f6cf-autoupdate.tgz

tar x -zvC / -f  openwrt-tailscale-enabler-v1.36.1-fb2f6cf-autoupdate.tgz

opkg update

opkg install libustream-openssl ca-bundle kmod-tun

/etc/init.d/tailscale start

/etc/init.d/tailscale enable
设置自启动

ls /etc/rc.d/S*tailscale*
验证成功会显示绿色

tailscale up --accept-dns=false --accept-routes --advertise-exit-node --advertise-routes=自己路由器的ip前三段.0/24

**Reply by ZRP1107** [reply]
安装步骤

**Reply by ZRP1107** [reply]
教程是错的，看看就行

**Reply by gzxiexl** [reply]
盼着有大佬做个GUI 
Unraid插件有Tailscale的GUI 不知道能否移植

**Reply by jjm2473** [reply]
有个docker版的，有网页可以控制，自己找找。vpn相关的东西不方便上架istore

**Reply by ZRP1107** [reply]
> 盼着有大佬做个GUI 
> Unraid插件有Tailscale的GUI 不知道能否移植

GL.inet有集成

**Comment by picc745514** [comment]
ARS2怎么不更新了？抛弃了吗？

**Reply by jjm2473** [reply]
ars2没啥好更新的，而且内核也太老，更新也没什么意义。其他机型，例如x86要升级内核才能支持新的核显
